### PR TITLE
Add http enable flag for Node

### PR DESCRIPTION
**Pull request by influx6**
PR adds a flag to the `params.NodeConfig` which enables us define the state whether the internal rpc server initiated by the `ethereum.Node` instance, except when the flag is enabled and also is set to false by default.

Related to issue #177.

### CODE REVIEW

File: geth/params/config.go [hunk 1]
```diff
@@ -64,6 +66,27 @@ type FirebaseConfig struct {
...
+		return nil, ErrEmptyAuthorizationKeyFile
+	}
+
+	key, err := ioutil.ReadFile(c.AuthorizationKeyFile)
+	if err != nil {
+		return nil, err
+	}
+	key = bytes.TrimRight(key, "\n")
+
+	if key == nil {
```
Comment (tiabc) [inline geth/params/config.go]: Let's check len here? Because of this: https://play.golang.org/p/lTztWCWRv7

File: geth/params/config.go [hunk 2]
```diff
@@ -64,6 +66,27 @@ type FirebaseConfig struct {
 	NotificationTriggerURL string
 }
 
+// ReadAuthorizationKeyFile reads and loads FCM authorization key
+func (c *FirebaseConfig) ReadAuthorizationKeyFile() ([]byte, error) {
+	if len(c.AuthorizationKeyFile) <= 0 {
+		return nil, ErrEmptyAuthorizationKeyFile
```
Comment (tiabc) [inline geth/params/config.go]: In my opinion, it's better to return other error here in order to faster distinguish if the authorization file is set or empty.

File: geth/node/node.go [hunk 3]
```diff
@@ -123,16 +123,24 @@ func defaultEmbeddedNodeConfig(config *params.NodeConfig) *node.Config {
...
+		// HTTPPort:    config.HTTPPort,
 		HTTPCors:    []string{"*"},
 		HTTPModules: strings.Split(config.APIModules, ","),
 		WSHost:      makeWSHost(config),
 		WSPort:      config.WSPort,
 		WSOrigins:   []string{"*"},
 		WSModules:   strings.Split(config.APIModules, ","),
 	}
+
+	if config.HTTPEnabledMode {
```
Comment (tiabc) [inline geth/node/node.go]: HTTPEnabledMode sounds a bit unclear to me. From one side, there's only 1 possible option implied by HTTPEnable and this is RPC. From the other side, HTTPEnabled is not the same as RPCEnabled and maybe we should consider using the second. What do you think?

File: geth/params/config.go (hunk 1 above)
Comment (influx6) [inline geth/params/config.go]: Sure. Sounds good.

File: geth/node/node.go (hunk 3 above)
Comment (influx6) [inline geth/node/node.go]: That should make sense and be more clearer in this context. I agree with that.

File: geth/params/config.go (hunk 2 above)
Comment (influx6) [inline geth/params/config.go]: Hmmmm, that sounds reasonable if we need such context that is. Will change it up.

File: geth/node/node.go (hunk 3 above)
Comment (jarradh) [inline geth/node/node.go]: maybe `ProxyEnabledMode`

Comment (tiabc) [inline geth/node/node.go]: But it has nothing to do with proxy, hasn't it?

Comment (jarradh) [inline geth/node/node.go]: or sorry, yes you're correct

File: geth/params/config.go [hunk 4]
```diff
@@ -107,12 +132,65 @@ type WhisperConfig struct {
 	FirebaseConfig *FirebaseConfig `json:"FirebaseConfig,"`
 }
 
+// ReadPasswordFile reads and returns content of the password file
+func (c *WhisperConfig) ReadPasswordFile() ([]byte, error) {
+	if len(c.PasswordFile) <= 0 {
+		return nil, ErrEmptyPasswordFile
```
Comment (tiabc) [inline geth/params/config.go]: Can you also fix this error to ErrPasswordFileNotSet, please?

File: geth/params/config.go [hunk 5]
```diff
@@ -107,12 +132,65 @@ type WhisperConfig struct {
...
+	}
+
+	return password, nil
+}
+
+// ReadIdentityFile reads and loads identity private key
+func (c *WhisperConfig) ReadIdentityFile() (*ecdsa.PrivateKey, error) {
+	if len(c.IdentityFile) <= 0 {
+		return nil, ErrEmptyIdentityFile
```
Comment (tiabc) [inline geth/params/config.go]: And here.

File: geth/params/config_test.go [hunk 6]
```diff
@@ -29,6 +29,7 @@ var loadConfigTestCases = []struct {
 			"WSPort": 8546,
 			"IPCEnabled": true,
 			"WSEnabled": false,
+			"HTTPEnabledMode": false,
```
Comment (tiabc) [inline geth/params/config_test.go]: The new flag is now named `RPCEnabledMode`, not `HTTPEnabledMode` and this setting has no effect as the default value is false. I'd rename it to the appropriate name for consistency here, below in this file and in other configs.
Could you also write a test with `"RPCEnabledMode": true` to be confident this flag is now working?

Comment (influx6) [inline geth/params/config_test.go]: Sure. Will do. Thanks, I missed that

**Pull request merged**